app/controllers/application_controller.rb

```diff
@@ -1,2 +1,21 @@
 class ApplicationController < ActionController::API
+  attr_reader :current_user
+
+  before_action :authenticate_user!
+
+  private
+
+  def authenticate_user!
+    authentication = RecognizeUser.call(token: auth_token)
+
+    if authentication.success?
+      @current_user = authentication.user
+    else
+      render json: { error: authentication.error }, status: :unauthorized
+    end
+  end
+
+  def auth_token
+    request.headers["Authorization"]&.split(" ")&.last
```

Comment:
Just for your information, you can use the `split` method here without passing any attributes. Example:
```bash
[1] pry(main)> str = "abc def"
=> "abc def"
[2] pry(main)> str.split
=> ["abc", "def"]
```

Your variant is also good!